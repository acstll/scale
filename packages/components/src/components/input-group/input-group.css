:host {
  display: flex;
  flex-direction: column;
}

.input-group {
  display: flex;
  flex-direction: inherit;
  width: 100%;
  margin-bottom: 1rem;
}
