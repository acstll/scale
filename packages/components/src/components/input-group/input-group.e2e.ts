import { newE2EPage } from '@stencil/core/testing';

describe('t-input-group', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<t-input-group/>');
    const element = await page.find('t-input-group');
    expect(element).toHaveClass('hydrated');
  });
});
