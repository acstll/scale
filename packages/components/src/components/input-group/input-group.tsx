import { Component, Prop, h } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';

@Component({
  tag: 't-input-group',
  styleUrl: 'input-group.css',
  shadow: true,
})
export class InputGroup {
  /** (optional) Input text class */
  @Prop() public customClass?: string = '';
  /** (optional) Input text theme */
  @Prop() public theme?: string = '';

  public render() {
    return (
      <div class={this.getCssClassMap()}>
        <slot />
      </div>
    );
  }

  private getCssClassMap(): CssClassMap {
    return classNames(
      'input-group',
      this.customClass && this.customClass,
      this.theme && `input-group--theme-${this.theme}`
    );
  }
}
