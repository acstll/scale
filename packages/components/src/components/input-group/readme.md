# t-input-text



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description                 | Type     | Default |
| ------------- | -------------- | --------------------------- | -------- | ------- |
| `customClass` | `custom-class` | (optional) Input text class | `string` | `''`    |
| `theme`       | `theme`        | (optional) Input text theme | `string` | `''`    |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
