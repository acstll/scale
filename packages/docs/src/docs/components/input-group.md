---
title: Input group
description: Input group element
url: /docs/input-group
contributors:
  - eeegor
---

# Input group

Input group component

selector: `t-input-group`

## Default

```javascript
<t-input-group>
	<t-input></t-input>
</t-input-group>
```

<div class="demo-container">
  <t-input-group>
		<t-input></t-input>
	</t-input-group>
</div>

## With label

```javascript
<t-input-group>
	<t-input-label>Name</t-input-label>
	<t-input></t-input>
</t-input-group>
```

<div class="demo-container">
  <t-input-group>
		<t-input-label>Name</t-input-label>
		<t-input></t-input>
	</t-input-group>
</div>

## With label and error

```javascript
<t-input-group>
	<t-input-label>Name</t-input-label>
	<t-input></t-input>
	<t-input-error>Required</t-input-error>
</t-input-group>
```

<div class="demo-container">
  <t-input-group>
		<t-input-label>Name</t-input-label>
		<t-input></t-input>
		<t-input-error>Required</t-input-error>
	</t-input-group>
</div>

## Multiple groups

```javascript
<form>
	<t-input-group>
		<t-input-label>Name</t-input-label>
		<t-input></t-input>
	</t-input-group>
	<t-input-group>
		<t-input-label>Password</t-input-label>
		<t-input type="password"></t-input>
	</t-input-group>
</form>
```

<div class="demo-container">
  <form>
		<t-input-group>
			<t-input-label>Name</t-input-label>
			<t-input></t-input>
		</t-input-group>
		<t-input-group>
			<t-input-label>Password</t-input-label>
			<t-input type="password"></t-input>
		</t-input-group>
	</form>
</div>
