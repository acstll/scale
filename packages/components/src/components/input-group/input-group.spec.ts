import { newSpecPage } from '@stencil/core/testing';
import { InputGroup } from './input-group';

describe('Input Group', () => {
  let element;
  beforeEach(async () => {
    element = new InputGroup();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [InputGroup],
      html: `<t-input-group>Label</t-input-group>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle theme css class', () => {
    element.theme = 'default';
    expect(element.getCssClassMap()).toContain('input-group--theme-default');
  });

  it('should have a default css class', () => {
    expect(element.getCssClassMap()).toContain('input-group');
  });
});
