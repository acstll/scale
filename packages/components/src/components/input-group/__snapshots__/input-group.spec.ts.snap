// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Group should match snapshot 1`] = `
<t-input-group>
  <mock:shadow-root>
    <div class="input-group">
      <slot></slot>
    </div>
  </mock:shadow-root>
  Label
</t-input-group>
`;
